controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java

```diff
@@ -106,6 +106,9 @@
     private final Curator curator;
     private final Duration tryLockTimeout;
 
+    // For each application id (path), store the ZK node version and its deserialised data - update when version changes.
```

Comment:
```suggestion
    // For each application id (path), store the ZK node version and its deserialised data - update when version changes.
    // This will grow to keep all applications in memory, but this should be OK
     ```